File: backend/identity-service/src/main/java/com/offerhub/identity/dto/ApiResponse.java
package com.offerhub.identity.dto;

import lombok.Getter;

@Getter
public class ApiResponse<T> {
    private final boolean success;
    private final T data;
    private final ApiError error;

    private ApiResponse(boolean success, T data, ApiError error) {
        this.success = success;
        this.data = data;
        this.error = error;
    }

    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>(true, data, null);
    }

    public static <T> ApiResponse<T> error(String code, String message) {
        return new ApiResponse<>(false, null, new ApiError(code, message));
    }

    public static <T> ApiResponse<T> error(String code, String message, java.time.Instant lockedUntil) {
        String lockedUntilStr = lockedUntil != null ? lockedUntil.toString() : null;
        return new ApiResponse<>(false, null, new ApiError(code, message, lockedUntilStr));
    }
}

File: mobile/OfferHub/app/src/main/java/com/example/offerhub/data/network/ApiResponse.kt
package com.example.offerhub.data.network

data class ApiResponse<T>(val success: Boolean, val data: T?, val error: ApiError?)

data class ApiError(
    val code: String,
    val message: String? = null,
    val lockedUntil: String? = null,
    val remainingSeconds: Long? = null
)

data class PagedResult<T>(
    val items: List<T>,
    val total: Long,
    val page: Int,
    val size: Int
)
